### ISSUE TITLE
The reader is not working.

### ISSUE THREAD

**Issue by reeshabhranjan** [Eromera/uah_driveset_reader#3]
I have tried running the given python script on Windows and Ubuntu 16.04. It is giving some error:

```
['20151126130708-16km-D3-AGGRESSIVE-SECONDARY.mp4']
(6785, 11)                                                                                                              True
(6785, 11)                                                                                                              True
(6785, 11)                                                                                                              True
(6785, 11)                                                                                                              True
Video loaded?: False
Data loaded?: True
```
Related video file: https://youtu.be/o3tvaDYY6ks

**Comment by highintothesky** [comment]
you need to install the following versions for the full program to work. the authors should have included this info:

`imageio==2.4.1`

`matplotlib==2.0.0`

**Comment by andersonfvinco** [comment]
The same error happened to me. After installing the package versions listed by @highintothesky, the reader now works fine.

**Comment by Drkramarao** [comment]
error in installing matplotlib==2.0.0

C:\WINDOWS\system32>pip install matplotlib==2.0.0
Collecting matplotlib==2.0.0
  Using cached matplotlib-2.0.0.tar.gz (53.2 MB)
    ERROR: Command errored out with exit status 1:
     command: 'c:\program files (x86)\python38-32\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-ee0k3y8e\\matplotlib\\setup.py'"'"'; __file__='"'"'C:\\Users\\User\\AppData\\Local\\Temp\\pip-install-ee0k3y8e\\matplotlib\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\User\AppData\Local\Temp\pip-pip-egg-info-w0kbv4v4'
         cwd: C:\Users\User\AppData\Local\Temp\pip-install-ee0k3y8e\matplotlib\
    Complete output (66 lines):
    ============================================================================
    Edit setup.cfg to change the build options

    BUILDING MATPLOTLIB
                matplotlib: yes [2.0.0]
                    python: yes [3.8.3 (tags/v3.8.3:6f8c832, May 13 2020,
                            22:20:19) [MSC v.1925 32 bit (Intel)]]
                  platform: yes [win32]

    REQUIRED DEPENDENCIES AND EXTENSIONS
                     numpy: yes [version 1.19.0]
                       six: yes [using six version 1.15.0]
                  dateutil: yes [using dateutil version 2.8.1]
               functools32: yes [Not required]
              subprocess32: yes [Not required]
                      pytz: yes [pytz was not found. pip will attempt to install
                            it after matplotlib.]
                    cycler: yes [using cycler version 0.10.0]
                   tornado: yes [tornado was not found. It is required for the
                            WebAgg backend. pip/easy_install may attempt to
                            install it after matplotlib.]
                 pyparsing: yes [using pyparsing version 2.4.7]
                    libagg: yes [pkg-config information for 'libagg' could not
                            be found. Using local copy.]
                  freetype: no  [The C/C++ header for freetype (ft2build.h)
                            could not be found.  You may need to install the
                            development package.]
                       png: no  [The C/C++ header for png (png.h) could not be
                            found.  You may need to install the development
                            package.]
                     qhull: yes [pkg-config information for 'qhull' could not be
                            found. Using local copy.]

    OPTIONAL SUBPACKAGES
               sample_data: yes [installing]
                  toolkits: yes [installing]
                     tests: no  [skipping due to configuration]
            toolkits_tests: no  [skipping due to configuration]

    OPTIONAL BACKEND EXTENSIONS
                    macosx: no  [Mac OS-X only]
                    qt5agg: yes [installing, Qt: 5.15.0, PyQt: 5.15.0]
                    qt4agg: no  [PySide not found; PyQt4 not found]
                   gtk3agg: no  [Requires pygobject to be installed.]
                 gtk3cairo: no  [Requires cairocffi or pycairo to be installed.]
                    gtkagg: no  [Requires pygtk]
                     tkagg: yes [installing; run-time loading from Python Tcl /
                            Tk]
                     wxagg: no  [requires wxPython]
                       gtk: no  [Requires pygtk]
                       agg: yes [installing]
                     cairo: no  [cairocffi or pycairo not found]
                 windowing: yes [installing]

    OPTIONAL LATEX DEPENDENCIES
                    dvipng: no
               ghostscript: no
                     latex: no
                   pdftops: no

    OPTIONAL PACKAGE DATA
                      dlls: no  [skipping due to configuration]

    ============================================================================
                            * The following required packages can not be built:
                            * freetype, png
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

**Comment by ChenxuWen-hub** [comment]
Thank you for your answer @highintothesky

**Comment by 18545031552** [comment]
> 同样的错误也发生在我身上。安装由您列出的包版本@highintothesky，读取器现在工作正常。

What is your version of Python

**Comment by 18545031552** [comment]
> 安装 matplotlib= 2.0.0 时出错
> 
> C：\WINDOWS\System32>pip
> 安装 matplotlib=2.0.0 收集 matplotlib= 2.0.0
> 使用缓存 matplotlib-2.0.0.tar.g z （53.2 MB）
> 错误： 命令
> 错误与退出状态 1： 命令： 'c：\ 程序文件 （x86）\ python38-32\python.exe' - c'导入系统， 设置工具， 标记;sys.argv=0= "C：\用户\User\AppData\本地\Temp\pip-安装-ee0k3y8e\matplotlib\ setup.py'"'"""""""""""""""""""""""""""""""""""""""""""""""用户"="用户"=""""用户=用户=本地=临时性=pip-安装-ee0k3y8e=matplotlib"=setup.py"""""""""""""""""""""""""""""""**文件**\''C：\用户\用户\应用程序\本地\临时\pip-安装-ee0k3y8e\matplotlib\setup.py'''';f\getattr''，'标记，''打开''，打开）（**文件**）;代码\f.read（. 替换（'''''r\n''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''\n''''）;f.close（）;执行（编译（代码，**文件**
> ，''''exec'egg_info'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''{Temp-pip-pip-egg-info-w0kbv4v4' cwd： C：\用户\用户\AppData\本地\Temp\pip-安装-ee0k3y8e\matplotlib
> 完成输出（66 行）：
> 
> [] [] 编辑设置. cfg 更改生成选项
> 
> ```
> BUILDING MATPLOTLIB
>             matplotlib: yes [2.0.0]
>                 python: yes [3.8.3 (tags/v3.8.3:6f8c832, May 13 2020,
>                         22:20:19) [MSC v.1925 32 bit (Intel)]]
>               platform: yes [win32]
> 
> REQUIRED DEPENDENCIES AND EXTENSIONS
>                  numpy: yes [version 1.19.0]
>                    six: yes [using six version 1.15.0]
>               dateutil: yes [using dateutil version 2.8.1]
>            functools32: yes [Not required]
>           subprocess32: yes [Not required]
>                   pytz: yes [pytz was not found. pip will attempt to install
>                         it after matplotlib.]
>                 cycler: yes [using cycler version 0.10.0]
>                tornado: yes [tornado was not found. It is required for the
>                         WebAgg backend. pip/easy_install may attempt to
>                         install it after matplotlib.]
>              pyparsing: yes [using pyparsing version 2.4.7]
>                 libagg: yes [pkg-config information for 'libagg' could not
>                         be found. Using local copy.]
>               freetype: no  [The C/C++ header for freetype (ft2build.h)
>                         could not be found.  You may need to install the
>                         development package.]
>                    png: no  [The C/C++ header for png (png.h) could not be
>                         found.  You may need to install the development
>                         package.]
>                  qhull: yes [pkg-config information for 'qhull' could not be
>                         found. Using local copy.]
> 
> OPTIONAL SUBPACKAGES
>            sample_data: yes [installing]
>               toolkits: yes [installing]
>                  tests: no  [skipping due to configuration]
>         toolkits_tests: no  [skipping due to configuration]
> 
> OPTIONAL BACKEND EXTENSIONS
>                 macosx: no  [Mac OS-X only]
>                 qt5agg: yes [installing, Qt: 5.15.0, PyQt: 5.15.0]
>                 qt4agg: no  [PySide not found; PyQt4 not found]
>                gtk3agg: no  [Requires pygobject to be installed.]
>              gtk3cairo: no  [Requires cairocffi or pycairo to be installed.]
>                 gtkagg: no  [Requires pygtk]
>                  tkagg: yes [installing; run-time loading from Python Tcl /
>                         Tk]
>                  wxagg: no  [requires wxPython]
>                    gtk: no  [Requires pygtk]
>                    agg: yes [installing]
>                  cairo: no  [cairocffi or pycairo not found]
>              windowing: yes [installing]
> 
> OPTIONAL LATEX DEPENDENCIES
>                 dvipng: no
>            ghostscript: no
>                  latex: no
>                pdftops: no
> 
> OPTIONAL PACKAGE DATA
>                   dlls: no  [skipping due to configuration]
> 
> ============================================================================
>                         * The following required packages can not be built:
>                         * freetype, png
> ----------------------------------------
> ```
> 
> 错误：命令在退出状态 1： python setup.py egg_info检查日志以获得完整的命令输出。

I have the same problem as you. Have you solved your problem

**Comment by Eromera** [comment]
Hi everyone. Thanks for creating the issue and solution. I have found that the best solution so far is to install matplotlib==2.2.5 and imageio==2.4.1. The matplotlib error is just a "ax.hold" deprecation, because this function was removed in matplotlib 3. Until I find time to update the code, using the last matplotlib 2 version (2.2.5) should work well. The imageio error is just due to the need to download some ffmpeg library and should also be easy to fix so it works for new versions.
